core/src/main/scala/cats/MonoidK.scala

```diff
@@ -67,8 +84,54 @@ import simulacrum.typeclass
     new ComposedMonoidK[F, G] {
       val F: MonoidK[F] = self
     }
+
+  /**
+   * Return `a` combined with itself `n` times.
+   *
+   * Example:
+   * {{{
+   * scala> SemigroupK[List].combineNK(List(1), 5)
+   * res0: List[Int] = List(1, 1, 1, 1, 1)
+
+   * scala> MonoidK[List].combineNK(List("ha"), 0)
+   * res1: List[String] = List()
+   *
+   * }}}
+   */
+  override def combineNK[A](a: F[A], n: Int): F[A] =
+    if (n < 0) throw new IllegalArgumentException("Repeated combining for monoids must have n >= 0")
+    else if (n == 0) empty[A]
+    else repeatedCombineNK(a, n)
+
+  /**
+   * Given a sequence of `as`, sum them using the monoidK and return the total.
+   *
+   * Example:
+   * {{{
+   * scala> MonoidK[List].combineAllK(List(List("One"), List("Two"), List("Three")))
+   * res0: List[String] = List(One, Two, Three)
+   *
+   * scala> MonoidK[List].combineAllK[String](List.empty)
+   * res1: List[String] = List()
+   * }}}
+   */
+  def combineAllK[A](as: IterableOnce[F[A]]): F[A] =
+    as.iterator.foldLeft(empty[A])(combineK[A])
+
+  override def combineAllOptionK[A](as: IterableOnce[F[A]]): Option[F[A]] =
```

Comment:
I think it is better to do:
```scala
def combineAllK(as: IterableOnce[F[A]]): F[A] =
  combineAllOptionK(as) match {
    case Some(fa) => fa
    case None => empty[A]
  }
```

and not override `combineAllOptionK`.

The reason is, you want to only have to override one thing when you implement MonoidK and not have to override both and since combineAllOptionK is on both MonoidK and SemigroupK I think it is the easier one to remember to override.

I have sadly seen many cases of people picking the wrong one to override and the code paths not hitting the optimized method.

Alternatively you could do it this way, but I make this override final (that's allowed right)? so no one accidentally makes the mistake.

Reply:
Thanks 👍